com/williamfiset/datastructures/binarysearchtree/SplayTree.java

```diff
@@ -0,0 +1,274 @@
+import com.williamfiset.datastructures.utils.TreePrinter;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+class BinaryTree<T extends Comparable<T>> implements TreePrinter.PrintableNode {
+    private BinaryTree<T> left, right;
+    private T data;
+
+    public BinaryTree(T data) {
+        this.data = data;
+    }
+
+    @Override
+    public BinaryTree<T> getLeft() {
+        return left;
+    }
+
+    @Override
+    public BinaryTree<T> getRight() {
+        return right;
+    }
+
+    @Override
+    public String getText() {
+        return data.toString();
+    }
+    public void setRight(BinaryTree<T> right) {
+        this.right = right;
+        //   right.parent = this;
+    }
+    public void setLeft(BinaryTree<T> left) {
+        this.left = left;
+        // left.parent = this;
+    }
+    public T getData() {
+        return data;
+    }
+
+    public void setData(T data) {
+        this.data = data;
+    }
+
+    @Override
+    public String toString() {
```

Comment:
This method should be in the SplayTree class with the node to display as the root so that someone can just do something like:

SplayTree mysplaytree = new SplayTree();
...
// Insert stuff
...
System.out.println(mysplaytree);

Reply:
Yes but since SplayTree extends BinaryTree it will call the toString from BinaryTree. 
so doing like 
System.out.println(mysplaytree); works fine.